### PR TITLE
src: lib: align folder organization with structure diagram.

### PR DESCRIPTION
**Pull request by aquilamacedo**
In this pull request (PR), our objective is to enhance code organization by relocating the library-related files to the `src/lib` folder. This change will greatly facilitate locating and accessing these files whenever required, ultimately streamlining the development process.

- [x] remote.sh
- [x] kwio.sh
- [x] kwlib.sh
- [x] kw_config_loader
- [x] kw_time_and_date.sh
- [x] statistics.sh
- [x] kw_include
- [x] signal_manager

Closes: #833

### CODE REVIEW

File: kw [hunk 1]
```diff
@@ -59,7 +59,7 @@ export KWORKFLOW
 # The include function (sourced from this file) should always be used for file sourcing.
 . "$KW_LIB_DIR/lib/kw_include.sh" --source-only
 
-include "$KW_LIB_DIR"/signal_manager.sh
+include "$KW_LIB_DIR"/lib/signal_manager.sh
```
Comment (magalilemes) [inline kw:62]: Use `include "${KW_LIB_DIR}/lib/signal_manager.sh"`, like in the other files.

File: kw [hunk 2]
```diff
@@ -57,7 +57,7 @@ fi
 export KWORKFLOW
 # This is the one and only time a file will be sourced this way.
 # The include function (sourced from this file) should always be used for file sourcing.
-. "$KW_LIB_DIR/kw_include.sh" --source-only
+. "$KW_LIB_DIR/lib/kw_include.sh" --source-only
```
Comment (magalilemes) [inline kw:60]: string concatenation style

File: kw [hunk 3]
```diff
@@ -117,15 +117,15 @@ function kw()
...
+        include "${KW_LIB_DIR}/lib/kw_config_loader.sh"
         include "${KW_LIB_DIR}/kw_ssh.sh"
 
         kw_ssh_main "$@"
       )
       ;;
     vars)
       (
-        include "$KW_LIB_DIR/kw_config_loader.sh"
+        include "$KW_LIB_DIR/lib/kw_config_loader.sh"
```
Comment (magalilemes) [inline kw:128]: string concatenation style

File: tests/kw_include_test.sh [hunk 4]
```diff
@@ -37,7 +37,7 @@ function test_include()
   local output
   local relpath
 
-  relpath="./src/kwio.sh"
+  relpath="./src/lib/kwio.sh"
```
Comment (magalilemes) [inline tests/kw_include_test.sh:40]: this could be single quotes

**Review by magalilemes — COMMENTED** [review]
LGTM. I only have a few nits on enforcing the string concatenation style.

**Comment by aquilamacedo** [comment]
Hi @magalilemes, thank you very much for your review. I just made the changes you asked for :)

**Comment by magalilemes** [comment]
Just pushed to the unstable branch. Thanks @aquilamacedo!

**Pull request closed without merging**